model/note/note.go

```diff
@@ -365,19 +365,42 @@ func get(inst *instance.Instance, file *vfs.FileDoc) (*Document, error) {
 	if doc := getFromCache(inst, file.ID()); doc != nil {
 		return doc, nil
 	}
-	// TODO replay steps
-	return fromMetadata(file)
+	version, _ := versionFromMetadata(file)
+	steps, err := getSteps(inst, file.ID(), version)
+	if err != nil && err != ErrTooOld && !couchdb.IsNoDatabaseError(err) {
+		return nil, err
+	}
+	doc, err := fromMetadata(file)
+	if err != nil {
+		return nil, err
+	}
+	if len(steps) == 0 {
+		return doc, nil
+	}
+	if version, ok := steps[0]["version"].(float64); ok {
```

Comment:
I don't understand why we recompute the doc's version here when we have one from the file's metadata. If they don't match, shouldn't we error out?

Reply:
Why should error out? We have the steps that can be used to rebuild the latest version of the document.

Reply:
My bad, I missed the fact that we use early return statements before and that if we reach this line, we don't have a version and need to recompute it from the steps.